src/components/FacetResults.js

```diff
@@ -135,6 +137,28 @@ export default class FacetResults extends React.Component<FacetResultsDefaultPro
     return false;
   }
 
+  /**
+   * Filter Buckets in Facet that are already applied to the search filters.
+   * Buckets of type tagcloud and timeseries do not need to be filtered.
+   */
+  filterFacetBuckets(facet: SearchFacet, type: string): SearchFacet {
+    if (type === 'tagcloud' || type === 'timeseries') {
+      return facet;
+    }
+    const facetFilters = this.context.searcher.state.facetFilters;
+    const facetFiltersMap: Map<string, FacetFilter> = new Map();
```

Comment:
It's probably not _super_ expensive to build this map for every facet we look at, but it might make sense to build it outside of here (e.g. in the `renderFacets()` method) and pass it in as a parameter to `filterFacetBuckets()`.